### PR TITLE
2846 Added opacity of 1 to MuiTableSortLabel icon

### LINKED ISSUE #11: TableSortLabel should not be hidden when mouse is not hovering on it
## Describe the desired behavior

When a `.MuiTableSortLabel` is not hovered on, the arrow should have the ~disabled color~ 
(edit: should be a "disabledBackground" color of Gray500@12% for light theme, and a Black200@24% for dark theme). 

When it is hovered or in-use, the label should be `text.secondary`

## Describe the current behavior

The arrow has an opacity 0 on it when not in use.

image

## Additional Context

This will be part of the Tables effort.

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes #11 

#### Changes proposed in this Pull Request:
- Changed opacity of icon to 1

#### Screenshots:

https://user-images.githubusercontent.com/25982779/150321290-a59b5b76-52d2-47e9-9810-19d11ac1e1b0.mov

#### To Test:
- yarn start:showcase

### CODE REVIEW

**Comment by ektasawant** [comment]
Few queries I have.
1. What is expected behaviour for dark theme?
2. In issue description it is mentioned `When it is hovered or in-use, the label should be text.secondary`. Is label showing correct color when it is hovered?

**Comment by huayunh** [comment]
Same comment as Ekta. You would also need to write the changelog for this fix.

**Comment by bkarambe** [comment]
> Few queries I have.
> 
> 1. What is expected behaviour for dark theme?
> 2. In issue description it is mentioned `When it is hovered or in-use, the label should be text.secondary`. Is label showing correct color when it is hovered?

Updated a color to text.secondary and made opacity same as dark theme

**Comment by ektasawant** [comment]
Changes looks good 👍

**Review by jeffvg — APPROVED** [review]

**Review by andrupu — APPROVED** [review]

**Comment by emclaug2** [comment]
@huayunh Do you know where the .36 opacity came from in this PR?  The `color` change looks fine just unsure of the opacity.

**Comment by huayunh** [comment]
I updated the issue to articulate my design intent

File: src/blueTheme.ts [hunk 1]
```diff
@@ -696,13 +696,13 @@ export const blueTheme: ThemeOptions = {
...
                     "& $icon": {
-                        color: BLUIColors.black[300],
+                        color: ThemeColors.text.secondary,
                         opacity: 1,
                     },
                 },
             },
             icon: {
-                opacity: 0.12,
+                opacity: 0.36,
```
Comment (emclaug2) [inline src/blueTheme.ts:705]:
```suggestion
                opacity: 1,
                color: Color(BLUIColors.gray[500])
                    .alpha(0.12)
                    .string()
```

Comment (emclaug2) [reply src/blueTheme.ts:705]: Looking at the issue, the Light theme needs Gray[500]@.12 opacity when not active, and dark theme requires Gray[500]@.24 opacity.

File: src/blueTheme.ts [hunk 2]
```diff
@@ -696,13 +696,14 @@ export const blueTheme: ThemeOptions = {
...
-                        color: BLUIColors.black[300],
+                        color: ThemeColors.text.secondary,
                         opacity: 1,
                     },
                 },
             },
             icon: {
-                opacity: 0.12,
+                opacity: 1,
+                color: Color(BLUIColors.gray[500]).alpha(0.12).string(),
```
Comment (emclaug2) [inline src/blueTheme.ts:706]:
```suggestion
                color: Color(BLUIColors.black[200]).alpha(0.12).string(),
```

File: src/blueDarkTheme.ts [hunk 3]
```diff
@@ -764,7 +764,8 @@ export const blueDarkTheme: ThemeOptions = {
                 },
             },
             icon: {
-                opacity: 0.36,
+                opacity: 1,
+                color: Color(BLUIColors.gray[500]).alpha(0.24).string()
```
Comment (emclaug2) [inline src/blueDarkTheme.ts:768]:
```suggestion
                color: Color(BLUIColors.black[200]).alpha(0.24).string()
```

Comment (joebochill) [reply src/blueDarkTheme.ts:768]: Should this be a reference to a theme color variable instead of hardcoded?

Comment (emclaug2) [reply src/blueDarkTheme.ts]: Yes, I agree with you Joe; we really should be trying to avoid using hard-coded styles when possible.
@bkarambe

**Comment by jeffvg** [comment]
looking at this again..with latest blueTheme the MuiTableSort was at first hard to identify on screen but my eyes are not so good....gray 500 opacity 1

image

**Comment by bkarambe** [comment]
> looking at this again..with latest blueTheme the MuiTableSort was at first hard to identify on screen but my eyes are not so good....gray 500 opacity 1
> 
The same color, we are using in the designs
> image
Screenshot 2022-01-27 at 10 55 31 AM

**Review by jeffvg — APPROVED** [review]
looks good matches design. Just need to change date in change log

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,7 @@
 ### Changed
 
 -   Updated MUI `<Tooltip>` font size to .75rem ([#8](https://github.com/brightlayer-ui/react-themes/issues/8)).
+-   Changed MUI `<TableSortLabel>` light theme hovered icon color to text secondary and non-hovered icon opacity to `0.36` ([#11](https://github.com/brightlayer-ui/react-themes/issues/11)).
```
Comment (joebochill) [inline CHANGELOG.md:12]:
```suggestion
-   Updated `<TableSortLabel>` styles ([#11](https://github.com/brightlayer-ui/react-themes/issues/11)).
```

**Pull request merged**